### PR TITLE
Make savior injectable into delegates

### PR DESCRIPTION
**Pull request by passsy**
Makes tests easier because the ActivityDelegate will now not work with a global `PresenterSavior` singleton.

### CODE REVIEW

File: thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/TiActivityDelegate.java [hunk 1]
```diff
@@ -75,17 +75,21 @@
 
     private final PresenterViewBinder<V> mViewBinder;
 
+    private TiPresenterSavior mSavior;
```
Comment (rberghegger) [inline thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/TiActivityDelegate.java]: Could be `final`.

File: thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/TiActivityDelegate.java [hunk 2]
```diff
@@ -75,17 +75,21 @@
...
+
     private TiViewProvider<V> mViewProvider;
 
     public TiActivityDelegate(final DelegatedTiActivity<P> activityProvider,
             final TiViewProvider<V> viewProvider,
             final TiPresenterProvider<P> presenterProvider,
-            final TiLoggingTagProvider logTag) {
+            final TiLoggingTagProvider logTag,
+            TiPresenterSavior savior) {
```
Comment (rberghegger) [inline thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/TiActivityDelegate.java]: Could be `final`.

File: thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/TiFragmentDelegate.java [hunk 3]
```diff
@@ -66,17 +66,21 @@
 
     private final PresenterViewBinder<V> mViewBinder;
 
+    private TiPresenterSavior mSavior;
```
Comment (rberghegger) [inline thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/TiFragmentDelegate.java]: Could be `final`.

File: thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/PresenterSavior.java [hunk 4]
```diff
@@ -33,36 +32,33 @@
...
     @Nullable
-    public TiPresenter recover(final String id) {
-        return mPresenters.get(id);
+    public TiPresenter recover(final String presenterId) {
+        return mPresenters.get(presenterId);
     }
 
-    public String safe(@NonNull final TiPresenter presenter) {
+    @Override
+    public String save(@NonNull final TiPresenter presenter) {
```
Comment (passsy) [inline thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/PresenterSavior.java]: 🙈 that typo that lived there for years

**Comment by rberghegger** [comment]
I would merge this PR. @StefMa do you agree?

**Pull request merged**